### DISCUSSION TITLE
Question: How to deal with many entries

### DISCUSSION THREAD

**Discussion by LMSSonos** [OrchardCMS/OrchardCore#6852] [Ideas]
I have currently a page which is an agenda of upcoming events. Each entry has a few fields like Date, Name, Location, etc

I created ContentItem with all the fields and added them to a bag. Now as things start to grow we get more and more entries and it get's really messy for the overview. 

I wanted to ask what is the best way to handle that kind of data in the admin? I was thinking to use an Excel file and then have a page where you can upload it and so I generate the content. Any better ideas?

**Comment by vitalybrandes** [comment]
> 
> 
> I have currently a page which is an agenda of upcoming events. Each entry has a few fields like Date, Name, Location, etc
> 
> I created ContentItem with all the fields and added them to a bag. Now as things start to grow we get more and more entries and it get's really messy for the overview.
> 
> I wanted to ask what is the best way to handle that kind of data in the admin? I was thinking to use an Excel file and then have a page where you can upload it and so I generate the content. Any better ideas?

Not sure what is your content, but see this:

https://datatables.net/

**Comment by LMSSonos** [comment]
@vitalybrandes thanks for the suggestion. That would be perfect. But I need to have this in the Orchard-Admin. So how can I use this there? Do I need to build a module?

**Comment by vitalybrandes** [comment]
> 
> 
> @vitalybrandes thanks for the suggestion. That would be perfect. But I need to have this in the Orchard-Admin. So how can I use this there? Do I need to build a module?

@LMSSonos 
I think yes.
Try /check this example, maybe can point you to right way
https://github.com/Yuexs/OrchardCoreExample/tree/master/src/Modules/OrchardCoreExample.DataTable.Module

**Comment by LMSSonos** [comment]
Thanks, that was more or less what i was looking for and the admin part is clear. But how do I display this in the frontend facing pages?

**Comment by vitalybrandes** [comment]
> Thanks, that was more or less what i was looking for and the admin part is clear. But how do I display this in the frontend facing pages?

@LMSSonos 
Use this:
https://datatables.net/
You can do it via form, ajax etc.
I am using this library in the FP with razor pages and it's amazing!

**Comment by sebastienros** [comment]
Maybe the issue here is that you are using a bag. It will have to display and edit all items in the same screen. Using a List instead would allow you to page items, and edit things one at a time.

**Reply by LMSSonos** [reply]
Thanks @sebastienros  - this was what I was looking for. I have 2 small questions to it

1) Is there a way to add a filter and/or sort. for instance to sort by title. Else it will be hard to find items if you have many
2) After adding a new item to the list, is there a way to get back to the list? Somehow I'm stuck on that screen.
